doc/reference/trd104-syscalls.md

```diff
@@ -459,7 +473,9 @@ Command instance (combination of major and minor identifier) specifies
 its failure variant and success variant. If userspace invokes a
 command on a peripheral that is not installed, the kernel returns a
 failure variant of `Failure`, with an associated error code of
-`NOSUPPORT`.
+`NOSUPPORT`. Therefore, command invocations that need to
```

Comment:
There are a few references to `NOSUPPORT`, including the current implementation:

https://github.com/tock/tock/blob/38c12029d76530847591aded6afe14008f49668e/kernel/src/driver.rs#L186-L193

Reply:
Yeah, I think there's a discrepancy between the code and the error values. The kernel should return `ErrorCode::NODEVICE` if there is no `Driver` implementation for the device ID, but a system call driver should return 'ErrorCode::NOSUPPORT` if it's an invalid system call ID.

At least, this is what the ErrorCode/ReturnCode values seem to say to me, unless we want to interpret a "device" as something else. Or we could just tweak this text and come up with a definition of NODEVICE that's distinct from NOSUPPORT and UNINSTALLED.